spaceship.zsh

```diff
@@ -244,6 +256,62 @@ spaceship_ruby_version() {
   echo -n "%{$reset_color%}"
 }
 
+# Swift
+# Show current version of Swift
+spaceship_swift_version() {
+
+  if [[ $SPACESHIP_SWIFT_SHOW_GLOBAL == true ]] ; then
+    if command -v swiftenv > /dev/null 2>&1; then
+      local swift_version=$(swiftenv version | sed 's/ .*//')
+    fi
+  elif [[ $SPACESHIP_SWIFT_SHOW_LOCAL == true ]] ; then
+    if command -v swiftenv > /dev/null 2>&1; then
+      if swiftenv version | grep ".swift-version" > /dev/null; then
+        local swift_version=$(swiftenv version | sed 's/ .*//')
+      fi
+    fi
+  fi
+
+  if [ -n "${swift_version}" ]; then
+    echo -n " %B${SPACESHIP_PREFIX_SWIFT}%b "
+    echo -n "%{$fg_bold[yellow]%}"
+    echo -n "${SPACESHIP_SWIFT_SYMBOL}  ${swift_version}"
+    echo -n "%{$reset_color%}"
+  fi
+}
+
+# Xcode
+# Show current version of Xcode
+spaceship_xcode_version() {
+
+  if [[ $SPACESHIP_SWIFT_SHOW_GLOBAL == true ]] ; then
+    if command -v xcenv > /dev/null 2>&1; then
```

Comment:
Here the same: just replace these duplicated conditional by single line on top of function:

```zsh
command -v xcenv > /dev/null 2>&1 || return
```